docs/RationaleFIRRTL.md

```diff
@@ -59,12 +59,55 @@ community that would benefit from adding support for these, we can do so.
 
 ## Naming
```

Comment:
I'd move the definition of _named_ earlier in the text. A lot of the text hinges on understanding what this means. It may be good to also contrast this with LLVM/MLIR where naming is more explicit, e.g., `%0` vs. `%foo`.